pkg/ioutils/concat.go

```diff
@@ -0,0 +1,123 @@
+package ioutils
+
+import (
+	"io"
+	"errors"
+)
+
+func min(x, y int) int {
+	if x < y {
+		return x
+	}
+	return y
+}
+
+func max(x, y int) int {
+	if x < y {
+		return y
+	}
+	return x
+}
```

Comment:
Is there a reason you didn't use the Go math libraries for these functions? (https://golang.org/pkg/math/)